wsclient/src/test/java/org/hpccsystems/ws/client/WSFileIOClientTest.java

```diff
@@ -67,45 +69,47 @@ public class WSFileIOClientTest extends BaseRemoteTest
     @Test
     public void copyFile() throws Exception
     {
-        Assume.assumeFalse("Test not valid on containerized HPCC environment", client.isTargetHPCCContainerized());
-        assumeTrue("Ignoring test 'copyFile' because HPCC-30117 is not fixed", HPCC_30117.equalsIgnoreCase("fixed"));
-
         String lzfile=System.currentTimeMillis() + "_csvtest.csv";
         String hpccfilename="temp::" + lzfile;
         client.createHPCCFile(lzfile, targetLZ, true);
         byte[] data = "Product,SKU,Color\r\nBike,1234,Blue\r\nCar,2345,Red\r\n".getBytes();
         client.writeHPCCFileData(data, lzfile, targetLZ, true, 0, 20);
         try
         {
-            ProgressResponseWrapper dfuspray=wsclient.getFileSprayClient().sprayVariable(
+            ProgressResponseWrapper dfuspray = wsclient.getFileSprayClient().sprayVariable(
                     new DelimitedDataOptions(),
                     wsclient.getFileSprayClient().fetchLocalDropZones().get(0),
                     lzfile,"~" + hpccfilename,"",thorClusterFileGroup,true,
                     HPCCFileSprayClient.SprayVariableFormat.DFUff_csv,
                     null, null, null, null, null, null, null);
-            Thread.sleep(1000);
-            int wait=60;
             if (dfuspray.getExceptions() != null
-                    && dfuspray.getExceptions().getException() != null
-                    && dfuspray.getExceptions().getException().size()>0)
+                && dfuspray.getExceptions().getException() != null
+                && dfuspray.getExceptions().getException().size()>0)
             {
                 fail(dfuspray.getExceptions().getException().get(0).getMessage());
             }
-            if (dfuspray.getSecsLeft()>0)
+
+            List<String> whiteListedStates = Arrays.asList( "queued", "started", "unknown", "finished", "monitoring");
+            int waitCount = 0;
+            int MAX_WAIT_COUNT = 60;
+
+            ProgressResponseWrapper dfuProgress = null;
+            do
             {
-                System.out.println("Still spraying, waiting 1 sec...");
-                for (int i=wait;i>0;i--)
+                dfuProgress = wsclient.getFileSprayClient().getDfuProgress(dfuspray.getWuid());
+                if (!whiteListedStates.contains(dfuProgress.getState())) {
+                    fail("File spray failed: Summary: " + dfuProgress.getSummaryMessage() + " Exceptions: " + dfuProgress.getExceptions());
+                }
+
+                if (dfuProgress.getPercentDone() < 100)
                 {
-                    if (dfuspray.getSecsLeft()==0)
-                    {
-                        i=0;
-                    }
-                    else
-                    {
-                        Thread.sleep(1000);
-                    }
+                    Thread.sleep(1000);
+                    System.out.println("Percent complete: " + dfuProgress.getPercentDone() + " Sleeping for 1sec to wait for spray.");
```

Comment:
Minor because we're not likely to see this message often, but when we do, it feels incomplete because we didn't announce we're spraying a file and I'm not sure it'll be obvious why we're reporting status for a spray.